### PR TITLE
Feature/get available rooms

### PR DESCRIPTION
**Pull request by jeroengrift**

### CODE REVIEW

File: src/com/capgemini/service/ReservationService.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package com.capgemini.service;
+
+import com.capgemini.data.ReservationRepository;
+import com.capgemini.data.RoomRepository;
+import com.capgemini.domain.Reservation;
+import com.capgemini.domain.Room;
+import com.capgemini.domain.RoomType;
+import java.util.*;
+
+//TODO: add Unit test
```
Comment (Thomvdm) [inline src/com/capgemini/service/ReservationService.java]: Remove

File: src/com/capgemini/service/ReservationService.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    private List<Room> getRoomsWithReservation(Date startDate, Date endDate) {
+        List<Room> allReservedRooms = new ArrayList();
+
+        for(Reservation reservation : reservationRepository.getAllReservations()){
+            Date reservationStart = reservation.getStartDate();
+            Date reservationEnd = reservation.getEndDate();
+
+//TODO: give an understandable name
```
Comment (Thomvdm) [inline src/com/capgemini/service/ReservationService.java]: checkone/two/three????

File: test/com/capgemini/service/ReservationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            //comparison, expected = 2 rooms left. compare with size of results
+            Assert.assertSame(2, result.size());
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void getAvailableRoomsInsideStartBeforeDateParametersTest() {
```
Comment (Thomvdm) [inline test/com/capgemini/service/ReservationServiceTest.java]: Can you clarify this method name?

File: test/com/capgemini/service/ReservationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            //comparison, expected = 0 rooms left. compare with size of results
+            Assert.assertSame(0, result.size());
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void getAvailableRoomsInsideEndAfterDateParametersTest() {
```
Comment (Thomvdm) [inline test/com/capgemini/service/ReservationServiceTest.java]: Can you clarify method name?

File: test/com/capgemini/service/ReservationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            //comparison, expected = 0 rooms left. compare with size of results
+            Assert.assertSame(0, result.size());
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void getAvailableRoomsWithRoomTypeOutside() {
```
Comment (Thomvdm) [inline test/com/capgemini/service/ReservationServiceTest.java]: Clarify method name?

File: test/com/capgemini/service/ReservationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            //comparison, expected = 1 rooms left. compare with size of results
+            Assert.assertSame(1, result.size());
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void getAvailableRoomsWithWrongRoomTypeOutside() {
```
Comment (Thomvdm) [inline test/com/capgemini/service/ReservationServiceTest.java]: Clarify method name?

**Review by Thomvdm — CHANGES_REQUESTED** [review]
Mooi man. Alleen ff methodetjes en TODOs verwijderen

**Review by jeroenknoester — APPROVED** [review]

**Pull request closed without merging**